src/main/scala/diplomacy/Resources.scala

```diff
@@ -208,6 +212,8 @@ class SimpleDevice(devname: String, devcompat: Seq[String]) extends Device
     val names = optDef("reg-names", named.map(x => ResourceString(DiplomacyUtils.regName(x._1).get)).toList) // names of the named address space
     val regs = optDef("reg", (named ++ bulk).flatMap(_._2.map(_.value)).toList) // address ranges of all spaces (named and bulk)
 
+    devNamePlusAddress = name
```

Comment:
Although doing it this way makes it a bit easier for you to write this code now, this will in the long run make the code much more complicated since you are relying on state being mutated at a specific point in time, and that means if you try accessing this value too early, you don't get the result that you want.

I think a better way of doing this would be to not rely on mutability and instead turn this mutable property into a purely functional method. The method should require the caller to pass in all the information that is needed to compute this value, and then it will not be a game of "find the needle in the haystack" for the next person to figure out when this property is set.

Reply:
Wes suggested mutability.

The caller does not have the info.

@terpstra ????

Reply:
I still stand by my statement that immutability will make this easier to read in the long run.

The caller does have the information. The caller has the resource bindings, which is the only thing you need to pass into the method I'm suggesting you create:

```scala
class SimpleDevice(devname: String, devcompat: Seq[String]) extends Device ... {
  ...

  def devNamePlusAddress(resources: ResourceBindings) = describeName(devname, resources)

  ...
}
```